partitura/utils/music.py

```diff
@@ -2560,19 +2557,25 @@ def performance_from_part(part, bpm=100, velocity=64):
     ----------
     part: Part
         The part from which we want to generate a performed part
-    bpm : float
-        Beats per minute
-    velocity: float or int
-        The MIDI velocity for all notes.
+    bpm : float, np.ndarray or callable
+        Beats per minute to generate the performance. If a the value is a float,
+        the performance will be generated with a constant tempo. If the value is
+        a np.ndarray, it has to be an array with two columns where the first
+        column is score time in beats and the second column is the tempo. If a
+        callable is given, the function is assumed to map score onsets in beats
+        to tempo values. Default is 100 bpm.
+    velocity: int, np.ndarray or callable
+        MIDI velocity of the performance. If a the value is an int, the
+        performance will be generated with a constant MIDI velocity. If the
+        value is a np.ndarray, it has to be an array with two columns where
+        the first column is score time in beats and the second column is the
+        MIDI velocity. If a callable is given, the function is assumed to map
+        score time in beats to MIDI velocity. Default is 64.
 
     Returns
     -------
     ppart: PerformedPart
-
-    Potential extensions
-    --------------------
-    * allow for bpm to be a callable or an 2D array with columns (onset, bpm)
```

Comment:
This pull request implements these extensions (now as part of the `performance_notearray_from_score_notearray` method below)